src/main/java/com/google/api/codegen/config/SampleSpec.java

```diff
@@ -70,48 +76,41 @@ public static boolean expressionMatchesId(String expression, String id) {
     return id.matches(expression);
   }
 
-  public Set<SampleValueSet> getMatchingValueSets(
+  public List<SampleValueSet> getMatchingValues(
       ClientMethodType methodForm, SampleType sampleType) {
-    return getMatchingValueSets(CallingForm.Generic, sampleType);
+    return getMatchingValues(CallingForm.Generic, sampleType);
   }
+
   /**
    * Returns the SampleValueSets that were specified for this methodForm and sampleType.
    *
    * @param methodForm The calling form for which value sets are requestd
    * @param sampleType The sample type for which value sets are requested
    * @return A set of SampleValueSets for methodForm andSampleType
    */
-  public Set<SampleValueSet> getMatchingValueSets(CallingForm methodForm, SampleType sampleType) {
-    Set<SampleValueSet> matchingValueSets = new LinkedHashSet<>();
-    List<SampleTypeConfiguration> sampleConfigList = getConfigFor(sampleType);
+  public List<SampleValueSet> getMatchingValues(CallingForm methodForm, SampleType sampleType) {
     String methodFormString = methodForm.toString();
 
-    for (SampleTypeConfiguration sampleConfig : sampleConfigList) {
-
-      // Determine whether sampleConfig applies to methodForm.
-      boolean configMatchesForm = false;
-      for (String callingFormExpression : sampleConfig.getCallingFormsList()) {
-        if (expressionMatchesId(callingFormExpression, methodFormString)) {
-          configMatchesForm = true;
-          break;
-        }
-      }
-      if (!configMatchesForm) {
-        continue;
-      }
-
-      // Add the value sets referenced in this sampleConfig.
-      for (String valueSetExpression : sampleConfig.getValueSetsList()) {
-        Iterable<String> valueSetNames =
-            Iterables.filter(
-                valueSets.keySet(), (String id) -> expressionMatchesId(valueSetExpression, id));
-        for (String name : valueSetNames) {
-          matchingValueSets.add(valueSets.get(name));
-        }
-      }
-    }
-
-    return matchingValueSets;
+    List<String> valueSetExpressions =
```

Comment:
Thanks for making this more concise. I suggest adding a comment to clarify the intent here, and maybe also below.

```
// Get the value set expressions configured for this calling form
```